rir/src/compiler/transform/insert_cast.cpp

```diff
@@ -9,10 +9,11 @@ pir::Instruction* InsertCast::cast(pir::Value* v, PirType t, Value* env) {
     if (v->type.maybePromiseWrapped() && !t.maybePromiseWrapped()) {
         return new pir::Force(v, env);
     }
-    if (v->type == RType::logical && t == NativeType::test) {
+    if (v->type == PirType(RType::logical).nonLazy() && t == NativeType::test) {
```

Comment:
this looks strange. Why would `PirType(RType::logical)` create a lazy value by default? you should probably fix the constructor instead